pkg/rpc/response/result/asset_state.go

```diff
@@ -43,3 +63,45 @@ func NewAssetState(a *state.Asset) AssetState {
 		IsFrozen:   a.IsFrozen,
 	}
 }
+
+// MarshalJSON implements json.Marshaler interface.
+func (a *AssetState) MarshalJSON() ([]byte, error) {
+	s := &assetState{
+		ID:         a.ID,
+		AssetType:  a.AssetType,
+		Name:       a.Name,
+		Amount:     a.Amount,
+		Available:  a.Available,
+		Precision:  a.Precision,
+		FeeMode:    a.FeeMode,
+		FeeAddress: a.FeeAddress.Reverse(), // FeeAddress should be marshaled in BE but default marshaler uses LE.
```

Comment:
Are you sure about that?

Reply:
It looks to me that it's always zero and there are no examples that could prove it being LE or BE in real life. Thus, I'd suggest using the default (which is LE) and not adding yet another wrapper.

Reply:
Actually, I'm not sure because official RPC API (https://docs.neo.org/docs/en-us/reference/rpc/latest-version/api/getassetstate.html) does not suppose serialization of fields `FeeAddress` and `FeeMode` (see also https://github.com/neo-project/neo/blob/master-2.x/neo/Ledger/AssetState.cs#L151).
In neo-go these fields are serialized, particularly, `FeeAddress` was serialized in BE, so I decided to leave it as it was before.

Reply:
Also fields `ID` and `AssetType` JSONify in `assetID` and `assetType` respectively, but in C# API it should be just `id` and `type`. Should I fix it?

Reply:
> It looks to me that it's always zero and there are no examples that could prove it being LE or BE in real life. Thus, I'd suggest using the default (which is LE) and not adding yet another wrapper.

Fixed

Reply:
> official RPC API does not suppose serialization of fields FeeAddress and FeeMode

We can drop them from the answer then.

> Should I fix it?

Yep, we should be 100% compatible in our outputs.

Reply:
Done